src/world.rs

```diff
@@ -774,6 +775,41 @@ impl World {
         self.archetypes_inner().iter()
     }
 
+    /// Faster way of accessing a Component many times
```

Comment:
This is *not* a faster way to access a single component. We should convey that it optimizes cases where you're performing repeated random access within a single component type, e.g.:

```suggestion
    /// Efficient random access within a single component type
```

We should also document the conditions under which a dynamic borrowck panic might arise.